react/AddressForm.tsx

```diff
@@ -1,46 +1,162 @@
+import rules from './countries/rules'
+import PlaceDetails from './PlaceDetails'
+import { FormattedMessage, defineMessages } from 'react-intl'
+import NumberOption from './components/NumberOption'
 
-interface Props {
-  address: Address
-  summary: LineFragment[][]
-}
+defineMessages({
+  country: {
+    defaultMessage: '',
+    id: 'place-components.label.country',
+  },
+  state: {
+    defaultMessage: '',
+    id: 'place-components.label.state',
+  },
+  city: {
+    defaultMessage: '',
+    id: 'place-components.label.city',
+  },
+  neighborhood: {
+    defaultMessage: '',
+    id: 'place-components.label.neighborhood',
+  },
+  ['street-road-avenue']: {
+    defaultMessage: '',
+    id: 'place-components.label.street-road-avenue',
+  },
+  ['number-option']: {
+    defaultMessage: '',
+    id: 'place-components.label.number-option',
+  },
+  complement: {
+    defaultMessage: '',
+    id: 'place-components.label.complement',
+  },
+  stateAbbreviation: {
+    defaultMessage: '',
+    id: 'place-components.label.stateAbbreviation',
+  },
+})
+
+const AddressForm: StorefrontFunctionComponent<{}> = () => {
+  const { address, setAddress } = useAddressContext()
+  const [editing, setEditing] = useState<boolean>(false)
+  const { fields, display } = rules[address.country]
+  const summary = display['extended'] as LineFragment[][]
+
+  const getSummaryFields = (summary: LineFragment[][]) => {
+    let summaryFields = new Set()
+    summary.forEach((line: LineFragment[]) => {
+      line.forEach((fragment: LineFragment) => {
+        summaryFields.add(fragment.name)
+      })
+    })
+    return summaryFields
+  }
+
+  const [ignoredFields, setIgnoredFields] = useState(
+    getSummaryFields(display['compact'])
+  )
+
+  const numberHasWithoutOption = (label: string | null) => {
+    return (
+      label && label.length >= 7 && label.substr(label.length - 7) == '-option'
+    )
+  }
 
-const AddressForm: StorefrontFunctionComponent<Props> = ({
-  address,
-  summary,
-}) => {
-  const { setAddress } = useAddressContext()
+  const hasOptions = (field: OptionsField) => {
+    return field && field.options
+  }
 
   const parseLineFragment = (fragment: LineFragment) => {
-    return address[fragment.name] != null ? (
+    const field = fields[fragment.name as keyof Fields]
+    const labelName = field && field.label ? field.label : null
+
+    if (ignoredFields.has(fragment.name)) return null
+    if (address[fragment.name] == null) return null
+    if (numberHasWithoutOption(labelName)) return <NumberOption showCheckbox />
+
+    const getFieldProps = (fragment: LineFragment) => {
+      const maxLength = field && field.maxLength ? field.maxLength : null
+      const autoComplete =
+        field && field.autoComplete ? field.autoComplete : null
+      const required = field && field.required ? field.required : null
+      const label = (
+        <FormattedMessage id={`place-components.label.${labelName}`} />
```

Comment:
An chance of `labelName` to be `null`? (according to https://github.com/vtex-apps/place-components/pull/5/files#diff-50aad4c7907d795061b7b081a887f18bR82)

Reply:
LabelName can't really be null, I just messed with typescript, and we already fixed this with @lucasecdb